src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -404,6 +409,8 @@ def prepare_create(self, daemon_spec: CephadmDaemonSpec) -> CephadmDaemonSpec:
         daemon_spec.ceph_conf = extra_config
         daemon_spec.keyring = keyring
 
+        daemon_spec.final_config, daemon_spec.deps = self.generate_config(daemon_spec)
+
```

Comment:
Calling `generate_config` from `prepare_create` makes sense to me, as they're basically the same function. What I don't understand is: why are we then **still** calling `generate_config` in  `_create_daemon()`:

https://github.com/ceph/ceph/blob/b659fef6d169e96bc6d21569be82003974a2c3a9/src/pybind/mgr/cephadm/serve.py#L768-L769

I think we should only have a **single** place where we call `generate_config` per daemon deployment.

Reply:
Yeah, this should not be here. I think I originally removed it then forgot to do so again when  rebasing after _create_daemon was moved to serve.py. I'll remove it again.